### ISSUE TITLE
How to process multiple tasks at the same time

### ISSUE THREAD

**Issue by OpenSource03** [clue/reactphp-mq#38]
Hello there!

I'm sorry for maybe asking such a beginner question, but I wasn't able to find a way to process multiple tasks at once.

For example, the following code does not work

```
\Clue\React\Mq\Queue::all(10, [1, 2, 3], function (int $number) {
    echo 'Processing: ', $number, PHP_EOL;
    sleep(10);
    return \React\Promise\resolve($number);
})->then(function (array $results) {
    echo 'Done: ', implode(', ', $results), PHP_EOL;
});
```

It seems to be processing each task separately. Is there a way to process all three of them at the same time?

Thanks

**Comment by bartvanhoutte** [comment]
You're calling `sleep` which is blocking the event loop. Use a non-blocking alternative. See [https://reactphp.org/promise-timer/#sleep](https://reactphp.org/promise-timer/#sleep) for more information.

**Comment by SimonFrings** [comment]
FYI: We recently released a new version of reactphp/async (`v4.1.0`, `v3.1.0` and `v2.1.0`) where we added a new `delay()` function to delay program execution. This new `delay()` function is the equivalent of `React\Async\await(React\Promise\Timer\sleep($seconds))` and much easier to use, so I'd recommend using it instead of https://reactphp.org/promise-timer/#sleep.

You can read more about it here: https://github.com/reactphp/async#delay
